iceoryx_hoofs/include/iceoryx_hoofs/internal/containers/uninitialized_array.hpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
+#define IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
+
+#include "iceoryx_hoofs/iceoryx_hoofs_types.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace containers
+{
+/// @brief struct used as policy parameter in UninitializedArray to wrap an array with all elements zeroed
+/// @tparam ElementType is the array type
+/// @tparam Capacity is the array size
+template <typename ElementType, uint64_t Capacity>
+struct ZeroedBuffer
+{
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : required by low level UninitializedArray building block and encapsulated in abstraction
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
+    using element_t = cxx::byte_t[sizeof(ElementType)];
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : required by low level UninitializedArray building block and encapsulated in abstraction
+    alignas(ElementType) element_t value[Capacity]{{0}};
+    // NOLINTEND(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
+};
+
+/// @brief struct used as policy parameter in UninitializedArray to wrap an uninitialized array
+/// @tparam ElementType is the array type
+/// @tparam Capacity is the array size
+template <typename ElementType, uint64_t Capacity>
+struct NotZeroedBuffer
```

Comment:
```suggestion
struct UninitializedBuffer
```

Reply:
IMO the `ZeroedBuffer` is still uninitialized in the way that no constructor of the underlying type is called. Therefore, I'd like to keep the name (i.e. I renamed it to `NonZeroedBuffer`).

Reply:
It is fine, but strictly speaking it could mean it is initialized to something but not zero. For me the important part is that there is no overhead proportional to the buffer size (setting all bytes to zero or something would have this overhead).

This was also the reason for the existence of the `Buffer` abstraction before.